api/server/models/user.js

```diff
@@ -0,0 +1,116 @@
+                            var data = {
+                                user_id: result.id,
+                            }
+                            callback(null, 1, 'Success login', data);
+                        } else if (action_type == 'manual_register') {
+                            callback(null, 0, 'Email id already exists.', data);
+                        } else if (action_type == 'manual_login') {
+                            var exist_password = result.password;
+                            GENERIC.match_encode_password(password, exist_password, function (check) {
+                                if (check == true) {
+                                    if (r_verification_status == 0) {
+                                        callback(null, 0, 'Please verify you account first', {});
+                                    } else {
+                                        var data = {
+                                            user_id: result.id,
+                                        }
+                                        callback(null, 1, 'Success login', data);
+                                    }
+                                } else {
+                                    callback(null, 0, 'Invalid Login', data);
+                                }
+                            })
+                        }
+                    } else {
+                        if (action_type == "facebook" || action_type == 'google') {
+                            password = GENERIC.get_random_number();
+                        }
+                        password = password.toString();
+                        GENERIC.encode_password(password, function (hash_password) {
+                            var verification_status = 0;
+                            var verification_code = GENERIC.get_random_number();
+                            verification_code = verification_code.toString();
+                            if (action_type == 'facebook' || action_type == 'google') {
+                                verification_status = 1;
+                                verification_code = '';
+                            }
+                            var newUser = new User({
+                                verification_status: verification_status,
+                                verification_code: verification_code,
+                                registration_type: action_type,
+                                social_id: social_id,
+                                platform: platform,
+                                device_id: device_id,
+                                token: token,
+                                email_id: email_id,
+                                name: name,
+                                password: hash_password,
+                                last_seen: '',
+                                registration_time: GENERIC.currentTimestamp(),
```

Comment:
this is not correct. ur getting server time instead of client time. 
client is in india and server is in germany. so this should wrong data.
the api needs to have timestamp sent in it